packages/transport-memory/src/index.js

```diff
@@ -1,83 +1,95 @@
 /**
  * Created by ozsayag on 26/06/2017.
  */
-import Promise from 'bluebird';
+import uuid from 'uuid/v1';
 
-import { QueueTransport } from '@stellarjs/messaging-queue';
+import { Transport } from '@stellarjs/transport';
+import get from 'lodash/get';
+import values from 'lodash/values';
+import { EventEmitter } from 'events';
 
-// standardise object to have json data spec
-function standardiseObject(obj) {
-  return JSON.parse(JSON.stringify(obj));
-}
-
-class MemoryTransport extends QueueTransport {
-  constructor(log) {
+class MemoryTransport extends Transport {
+  constructor(log, standardiseDates = false) {
     super(log);
-    this.queues = {};
-    this.subscribers = {};
+    this.subscriptionHandler = new EventEmitter();
+    this.standardiseDates = standardiseDates;
   }
 
-  setQueue(name, callback) {
-    if (!this.queues[name]) {
-      this.queues[name] = { currentId: 0 };
+// standardise object to have json data spec
+  standardiseObject(obj) {
+    if (this.standardiseDates) {
+      return JSON.parse(JSON.stringify(obj));
     }
 
-    if (callback) {
-      this.queues[name].callback = callback;
-    }
+    return obj;
   }
 
-  generateId(queueName) {
-    this.setQueue(queueName);
-    this.queues[queueName].currentId += 1;
-    return `${queueName}:${this.queues[queueName].currentId}`;
+  getLocalHandler(req) {
+    const url = get(req, 'headers.queueName');
```

Comment:
maybe it's a good timing to get rid of the `queue` term in stellar core.
Alternatives: `channel`/`subject`/`topic`